open-api/rest-catalog-open-api.yaml

```diff
@@ -1435,12 +1463,66 @@ components:
           additionalProperties:
             type: string
 
+    TokenRequest:
+      description:
+        OAuth2 client credentials or token exchange request
+
+
+        See -
+
+        - Client credentials flow - https://datatracker.ietf.org/doc/html/rfc6749#section-4.4
+
+        - Token exchange flow - https://datatracker.ietf.org/doc/html/rfc8693
+      type: object
+      required:
+        - grant_type
+      properties:
+        grant_type:
+          type: string
+          enum:
+            - client_credentials
+            - urn:ietf:params:oauth:grant-type:token-exchange
```

Comment:
Should we consider giving the prefix `urn:ietf:params:oauth:` special status, like via a reusable component? Maybe a schema element?

It can be used with `token-type`, `grant-type`, and many other things.

Maybe it could be  a prefix in an `OAuthParams` schema object that has various? various possible enums in it. Mostly thinking out loud.

Reply:
Splitting it would make this more confusing, I think. There's not a lot to reuse here besides a prefix.

Reply:
Fair point.